test/forcing/TrivialForcingProvider.hpp

```diff
@@ -3,45 +3,49 @@
 
 #include <string>
 #include <vector>
-#include "ForcingProvider.hpp"
+#include "GenericDataProvider.hpp"
+
 
 #define OUTPUT_NAME_1 "output_name_1"
 #define OUTPUT_VALUE_1 42.0
 #define OUTPUT_DEFAULT_1 36.8
 
 using namespace std;
 
-namespace forcing {
+namespace data_access {
     namespace test {
 /**
  * A trivial implementation, strictly for testing OptionalWrappedProvider.
  */
-        class TrivialForcingProvider : public ForcingProvider {
+        class TrivialForcingProvider : public GenericDataProvider {
         public:
 
             TrivialForcingProvider() {
                 outputs.push_back(OUTPUT_NAME_1);
             }
-
-            const vector<string> &get_available_forcing_outputs() override {
+                                  
+            const std::vector<std::string>& get_avaliable_variable_names() override {
                 return outputs;
             }
 
-            time_t get_forcing_output_time_begin(const string &output_name) override {
+            long get_data_start_time() override {
                 return 0;
```

Comment:
Actually this should probably be `std::numeric_limits<long>::min()` (negative) for the same reason we changed this to `long` instead of `time_t`, and `get_data_stop_time()` below should probably be `std::numeric_limits<long>::max()`